ci/cron/src/Main.hs

```diff
@@ -325,43 +238,62 @@ fetch_gh_paginated url = do
                 (_, _, _, [url, rel]) -> (rel, url)
                 _ -> fail $ "Assumption violated: link header entry did not match regex.\nEntry: " <> l
 
-fetch_gh_versions :: IO ([GitHubVersion], GitHubVersion)
+data Versions = Versions { top :: String, all :: Set.Set String, dropdown :: [String] }
```

Comment:
I would recommend to use `SemVer.Version` here instead of `String` so you don’t have to parse it everytime you use it.

Reply:
SemVer.Version is a bit unwieldy (and imo dangerous in this case), but I did add more types so conversion to String happens later. Hope that's what you had in mind.

I even turned some `String` strings into `FilePath` as human-reader-hint, though that one doesn't add safety.